### ISSUE TITLE
setBreak()

### ISSUE THREAD

**Issue by Georgelopes21** [rkl099/Appinventor-SerialOTGp3#1]
Could you please add setBreak(), setBreakon() duration, and setBreakoff() duration I tried to make an extension but it didn't work out

**Comment by rkl099** [comment]
I am curious, why do you need the break functions?

**Comment by Georgelopes21** [comment]
Hi rkl099,

I am trying to talk to a car ecu and I need to send a 5 baud x01 to start
communicating, that was why I asked if you could add FTDI bitbang support
but with setBreak() or sendBreak() I can use any serial devive not only
FTDI.
If you don't want to add this to your revision 3 Extention, I understand
that it's a very specific what I need, if there is any chance that you
could add SetBreak and FTDI bitbang on an extension and email it to me I
would be so grateful to you, I need that to talk to the ecu.

Thank you
George

**Comment by Georgelopes21** [comment]
HI

About the FTDI Bitbang mode, I had an idea to use that, to try to make a IR
Transmitter and RF Transmitter I know ftdi232 has a bug it is not very good
for ir and rf but I would like to try.

By the way, your Extention is fantastic I was able to bit-bang with the DTR
line but I need TX for the ECU.

**Comment by rkl099** [comment]
Ok, I think I have the information (from Linux src)  I need to make a test with setBreak() setBreakoff(). Some questions:
Can you use rev 1.1 of the extension instead of p3? (you have to do some updating of the blocks)
Can you handle break on timing with the AI2 timer?

**Comment by Georgelopes21** [comment]
Yes, rev 1.1 will work fine, I can handle the on-off with the clock timer
it worked fine with the DTR line.

**Comment by rkl099** [comment]
Maybe you can try this:

Lower the bit rate ex: 9600 to 4800 or 2400
Send 0x00
Return the bit rate to 9600 (or what you are using)

Bit rate can be changed when the connection is open, but you may have to insert a delay before returning to the original bit rate.

**Comment by Georgelopes21** [comment]
4800 or 2400 will never work it needs to be sent at 5 baud the only way is with ftdi d2xx, ft232 bitbang mode or with setBreak(), setBreakon() duration, and setBreakoff() duration.

**Comment by Georgelopes21** [comment]
The problem with physicaloid-library is that setBreak(), is not implemented a better choice would be to use https://github.com/mik3y/usb-serial-for-android it supports it for all chips, but that involves a total rewrite I understand that. you said in your initial tests that you used FTDI d2xx and it worked well, is there a chance that you could share that extension with me so i can test if it is possible to bit-bang the tx line from app inventor.

**Comment by Georgelopes21** [comment]
I just want to add your extension is fantastic I can do a lot with it and it works great but with custom hardware but I want to test it, on, off-the-shelf hardware.

**Comment by rkl099** [comment]
I have created a director Appinventor-d2xx with some notes on how to include d2xx in appinventor.
Please note: you have to build your own appinventor and do changes i physicaloid library to be able to use FTDI specific functions. This work only adds d2xx to the built-in serial component.
